### ISSUE TITLE
[BUG]: Silent Hill 2 cutscene broken/not rendering (Vulkan, D3D11)

### ISSUE THREAD

**Issue by DonutWrecker** [PCSX2/pcsx2#6976]
### Describe the Bug

In Silent Hill 2 Greatest Hits (USA) version, a cutscene mid-game does not render: Sound keeps playing, but whenever the pre-rendered/animated parts of the cutscene plays, screen goes black. When I switch the rendering mode to Software, the problem disappears, but present in D3D11 and Vulkan (haven't tried other rendering options) rendering modes.

### Reproduction Steps

1. Set the rendering options according to the images below.
<img width="599" alt="1" src="https://user-images.githubusercontent.com/67188697/187794741-49fab3ea-8c0b-4122-8bd3-ad534cb7b2c3.PNG">
<img width="600" alt="2" src="https://user-images.githubusercontent.com/67188697/187794743-ed757157-3250-47c7-a2be-ee000bb46a1a.PNG">

2. Load the game: Silent Hill 2 Greatest Hits (USA)
3. Load the savestate: 
[SLUS-20228 (FE06A030).01.p2s.zip](https://github.com/PCSX2/pcsx2/files/9465449/SLUS-20228.FE06A030.01.p2s.zip)

4. You will see the main character standing in front of a hole. Jump in there by pressing (X) button (Dualshock 4).
5. The cutscene will play.

### Expected Behavior

When a pre-rendered/animated scene plays, the screen will go black, the sound will keep playing. This will happen several times in [that cutscene](https://www.youtube.com/watch?v=9_P-V4JqAFg).

### PCSX2 Revision

v1.7.3269

### Operating System

Windows 10 (64bit)

### CPU

Ryzen 3 3100

### GPU

GTX 1650

### GS Settings

<img width="599" alt="1" src="https://user-images.githubusercontent.com/67188697/187794741-49fab3ea-8c0b-4122-8bd3-ad534cb7b2c3.PNG">
<img width="600" alt="2" src="https://user-images.githubusercontent.com/67188697/187794743-ed757157-3250-47c7-a2be-ee000bb46a1a.PNG">

### Emulation Settings

<img width="599" alt="3" src="https://user-images.githubusercontent.com/67188697/187795143-30d98645-ba1f-4602-bf34-1c836f0fe15f.PNG">

**Comment by refractionpcsx2** [comment]
try disabling MTVU and Instant VU.  if that's not enough try also enabling Fast CDVD.

This is kind of a known issue, SH2 sucks for this.

**Comment by DonutWrecker** [comment]
> try disabling MTVU and Instant VU. if that's not enough try also enabling Fast CDVD.
> 
> This is kind of a known issue, SH2 sucks for this.

None of them worked. I've already completed the game on PCSX2 1.6.0 stable build a year ago without any issues. I've also encountered the same problem a few months back while playing Silent Hill Origins; switched to the stable build afterwards by loading a hard save, just to watch the cut-scene. Kinda ruins the immersion, but it's the best way to experience the series today, so...

**Comment by felixthecat1970** [comment]
past year i create a workaround fix code to this issue, meanwhile: here reworked partial code:
```
gametitle=Silent Hill 2 (USA) (En,Ja,Fr,De,Es,It) (v2.01) SLUS_202.28;1 Game CRC = 0xFE06A030
comment=display fmv in HW mode

patch=0,EE,2027FAC4,extended,00000000 //wr buffer
patch=0,EE,2027FB9C,extended,00000000
patch=0,EE,2027FE74,extended,00000000 //fmv maybe buffer swap f. ?¿

```
bugs looks like is related to GS DISPLAY2 and ?¿ function related to frame swap (i guessing) when the fmv related func. build the second buffer 1080, disabling function call bufferswap and wr. regs 1080 val it leave default 1000 val buffer and FMV will show in HW, same case with SH3 .
about fmv screen position can be fixed by overscan option but is buggy in this game, flicker random glitches in main game screen (PCSX2 bug)

**Comment by BusterTheBusta** [comment]
I have no idea what that means or how to implement what you wrote, but one quick workaround I just found is going to "game fix" in the settings and then enabling "use Software Renderer for FMVs".

**Comment by refractionpcsx2** [comment]
well if the problem is the same as SH3, that means that the reason it fails is because the hw renderer doesn't know how to handle it.  The game starts the draw of the FMV right at the end of memory, so it loops around, then points the screen at the beginning of memory for the actual screen, but of course the hw renderer just doesn't understand at all, and you end up with a blank screen.

But yes software mode for FMV's is the fix for now, and should probably be added to this version of the game, if that's the case (not all of them seem to suffer from this)

**Comment by DonutWrecker** [comment]
> I have no idea what that means or how to implement what you wrote, but one quick workaround I just found is going to "game fix" in the settings and then enabling "use Software Renderer for FMVs".

I was able to apply that (use Software Renderer for FMVs) by [pressing the F9 key manually](https://forums.pcsx2.net/Thread-Where-is-the-switch-to-software-renderer-when-an-FMV-play), since I couldn't find it in the settings (v1.7.3275, which is the latest build by the time I'm writing this). Even though pressing the F9 key manually to turn on and off that setting is not practical, seems to be solved the issue.

Thanks everyone for the replies.

**Comment by refractionpcsx2** [comment]
The setting is under View->Properties while the game is running, then the "Game fixes" tab.

**Comment by DonutWrecker** [comment]
> The setting is under View->Properties while the game is running, then the "Game fixes" tab.

Oh, I didn't know that. Works almost perfectly, thanks.

**Comment by BusterTheBusta** [comment]
Weird. I finished Origins earlier this year on a 1.7 nightly build and I didn't find any problems. I switched to it in the middle of my playthrough because it fixed a bug in hardware mode with the not fully rendered flashlight. Don't know what version it was then unfortunately, and I have updated mine now, but afaik that version was WxWidgets only and I don't remember a Vulkan option

**Comment by refractionpcsx2** [comment]
Wait, is this a regression? or has it never worked properly?

**Comment by DonutWrecker** [comment]
> Wait, is this a regression? or has it never worked properly?

I think I've misremembered my problem with the Origins. I'm now pretty sure, that [when the cutscene switched to FMV](https://youtu.be/fYKy060tOrU?t=40), the game would stuck, and I couldn't even continue. I'm sorry, it's my bad.

I was able to recover the save regarding to [the moment I've had problem with](https://www.youtube.com/watch?v=fYKy060tOrU): looks like the problem has gone completely, even without the "use Software Renderer for FMVs" option unchecked. It now continues playing as it should.
